src/main/java/org/marketdesignresearch/mechlib/mechanism/auctions/Auction.java

```diff
@@ -1,61 +1,91 @@
 package org.marketdesignresearch.mechlib.mechanism.auctions;
 
-import com.google.common.base.Preconditions;
-import lombok.*;
-import lombok.extern.slf4j.Slf4j;
-import org.marketdesignresearch.mechlib.core.*;
-import org.marketdesignresearch.mechlib.core.bid.Bid;
-import org.marketdesignresearch.mechlib.core.bid.Bids;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+import org.marketdesignresearch.mechlib.core.Domain;
+import org.marketdesignresearch.mechlib.core.Good;
+import org.marketdesignresearch.mechlib.core.Outcome;
+import org.marketdesignresearch.mechlib.core.bid.bundle.BundleValueBids;
 import org.marketdesignresearch.mechlib.core.bidder.Bidder;
-import org.marketdesignresearch.mechlib.core.price.Prices;
-import org.marketdesignresearch.mechlib.instrumentation.AuctionInstrumentation;
-import org.marketdesignresearch.mechlib.mechanism.Mechanism;
 import org.marketdesignresearch.mechlib.instrumentation.AuctionInstrumentationable;
 import org.marketdesignresearch.mechlib.instrumentation.MipInstrumentation;
-import org.marketdesignresearch.mechlib.core.Outcome;
+import org.marketdesignresearch.mechlib.mechanism.Mechanism;
+import org.marketdesignresearch.mechlib.mechanism.auctions.interactions.Interaction;
 import org.marketdesignresearch.mechlib.outcomerules.OutcomeRuleGenerator;
 import org.springframework.data.annotation.PersistenceConstructor;
 
-import java.util.*;
-import java.util.stream.Collectors;
+import com.google.common.base.Preconditions;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
 
-@ToString(callSuper = true) @EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true) 
+@EqualsAndHashCode(callSuper = true)
 @Slf4j
-public class Auction extends Mechanism implements AuctionInstrumentationable {
+public abstract class Auction<BB extends BundleValueBids<?>> extends Mechanism implements AuctionInstrumentationable {
 
-    private static int DEFAULT_MAX_BIDS = 100;
-    private static int DEFAULT_MANUAL_BIDS = 0;
-    private static int DEFAULT_MAX_ROUNDS = 1;
+    private static int DEFAULT_MAX_ROUNDS = 0;
```

Comment:
I changed the default max rounds to infinite. Normally the auction will end when all AuctionPhase have ended.